Build failure with xml=no

Investigating how to create lightweight export templates, I'm trying to build it with as little options as possible.

It looks like ```xml=no``` is not a viable option currently, as the build fails with:
```
g++ -o core/register_core_types.x11.opt.64.o -c -g2 -fomit-frame-pointer -DFREETYPE_ENABLED -DOPENGL_ENABLED -DGLEW_ENABLED -DALSA_ENABLED -DPULSEAUDIO_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DGLES2_ENABLED -DGLES_OVER_GL -DPNG_ENABLED -D_3D_DISABLED -DGDSCRIPT_ENABLED -D_REENTRANT -Icore -Icore/math -Itools -Idrivers -I. -Iplatform/x11 -I/usr/include/freetype2 -I/usr/include/libpng16 -Icore/vorbis core/register_core_types.cpp
core/register_core_types.cpp: In function 'void register_core_types()':
core/register_core_types.cpp:104:2: error: 'resource_saver_xml' was not declared in this scope
  resource_saver_xml = memnew( ResourceFormatSaverXML );
  ^
core/register_core_types.cpp:106:2: error: 'resource_loader_xml' was not declared in this scope
  resource_loader_xml = memnew( ResourceFormatLoaderXML );
```

I started the build with:
```
scons -j3 target=release platform=x11 bits=64 tools=no vorbis=no minizip=no squish=no theora=no use_theoraplayer_binary=no freetype=no speex=no xml=no jpg=no webp=no dds=no pvr=no etc1=no openssl=no musepack=no disable_3d=yes
```

--- a/core/register_core_types.cpp
+++ b/core/register_core_types.cpp
@@ -101,10 +101,12 @@ void register_core_types() {
 	resource_loader_binary = memnew( ResourceFormatLoaderBinary );
 	ResourceLoader::add_resource_format_loader(resource_loader_binary);
 
+#ifdef XML_ENABLED
 	resource_saver_xml = memnew( ResourceFormatSaverXML );
 	ResourceSaver::add_resource_format_saver(resource_saver_xml);
 	resource_loader_xml = memnew( ResourceFormatLoaderXML );
 	ResourceLoader::add_resource_format_loader(resource_loader_xml);
+#endif
 
 	ObjectTypeDB::register_type<Object>();
 

